Makefile

```diff
@@ -203,6 +203,11 @@ install:
 
 	install -Dm644 -t ${PREFIX}/share/man/man1 ./docs/*.1
 
+	mkdir -p ${PREFIX}/include/hyprland
+	find src -name '*.h*' -exec cp --parents '{}' ${PREFIX}/include/hyprland ';'
+	cp ./*-protocol.h ${PREFIX}/include
```

Comment:
Not sure whether this is a good idea, we're kinda trashing `/usr/include`.
Wouldn't this be better put in like e.g. `${PREFIX}/include/hyprland/protocols`?

Reply:
Yeah, probably a good idea. I'll see if I can also make that work in meson.

Reply:
In general, IMHO, a clean solution is to move protocols to e.g. `/protocols/build`

Reply:
I tried this in Meson, but it doesn't work. Maybe someone has a better idea.

```diff
diff --git a/meson.build b/meson.build
index ddf5c97f..d7bc1a52 100644
--- a/meson.build
+++ b/meson.build
@@ -88,6 +88,12 @@ subdir('assets')
 subdir('example')
 subdir('docs')
 
+protocols_globber = run_command('find', '$MESON_BUILD_ROOT/protocols', '-name', '*.h*', check: true)
+protocols_headers = protocols_globber.stdout().strip().split('\n')
+foreach file : protocols_headers
+  install_headers(file, subdir: 'hyprland/protocols')
+endforeach
+
 pkg_install_dir = join_paths(get_option('datadir'), 'pkgconfig')
 
 import('pkgconfig').generate(
diff --git a/protocols/meson.build b/protocols/meson.build
index 193d7aaf..09003b7c 100644
--- a/protocols/meson.build
+++ b/protocols/meson.build
@@ -47,8 +47,6 @@ foreach p : protocols
 	wl_protos_headers += custom_target(
 		xml.underscorify() + '_server_h',
 		input: xml,
-		install: true,
-		install_dir: 'include',
 		output: '@BASENAME@-protocol.h',
 		command: [wayland_scanner, 'server-header', '@INPUT@', '@OUTPUT@'],
 	)
```

```
meson.build:91:0: ERROR: Command "/nix/store/j5wraaxv16fcl10x11566a3807nr4nlr-findutils-4.9.0/bin/find $MESON_BUILD_ROOT/protocols -name *.h*" failed with status 1.
```

Reply:
I'm very confused... Shouldn't you just change the install_dir?

Reply:
That was my first idea as well. Turns out Nix doesn't like that. The way it constructs the `hyprland-dev` package is by `mv`ing the include dir into another. And it fails when I change `install_dir` to any subdir of `include`, because the dir already exists and mv fails.
I will search for a workaround tomorrow.

Reply:
Shouldn't you just update Nix as well? To adapt to the updated install requirements.

Reply:
Actually this is probably a meson.build bug, you should respect -Dincludedir by using get_option() and then nix can use standard rulesets to install without manual mv.

Reply:
Nix relies on Meson & Ninja to do the heavy lifting of building the package and installing it into a directory. I did add a "dev" output which is what causes this behavior. I could leave it out though, and it shouldn't fail anymore.

Reply:
a possible workaround would be to patch meson to install into `include/` in nix only (and install into `include/hyprland` on other systems), that is if no other solution exists ofc

Reply:
@eli-schwartz your idea worked perfectly. Thanks!